packages/plugin-data/src/api/data/tree/exportApi.ts

```diff
@@ -0,0 +1,607 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+
+import * as path from 'path';
+import { fs, Logger, Messages, Org, SfdxError } from '@salesforce/core';
+import { AnyArray, AnyJson, get, getNumber, getString, has, isArray } from '@salesforce/ts-types';
+import { UX } from '@salesforce/command';
+import { sequentialExecute } from './executors';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/plugin-data', 'exportApi');
+
+const DATA_PLAN_FILENAME_PART = '-plan.json';
+
+const describe: any = {}; // holds metadata result for object type describe calls
+
+interface DataPlanPart {
+  sobject: string;
+  saveRefs: boolean;
+  resolveRefs: boolean;
+  files: string[];
+}
+
+export interface ExportConfig {
+  query: string;
+  outputDir?: string;
+  plan?: boolean;
+  prefix?: string;
+}
+
+/**
+ * Exports data from an org into sObject tree format.
+ */
+export class ExportApi {
+  private logger: Logger;
+
+  private objectTypeRegistry: any = {}; // registry for object type data plan descriptor
+  private referenceRegistry: any = {}; // registry of object type-specific id-to-ref mappings
+  private typeRefIndexes: any = {}; // registry for object type-specific ref counters
+
+  private config!: ExportConfig;
+
+  public constructor(private readonly org: Org, private readonly ux: UX) {
+    this.logger = Logger.childFromRoot(this.constructor.name);
+  }
+
+  /**
+   * Invokes the provided SOQL query against a target Org.  Results
+   * are converted into SObject Tree format.
+   *
+   * @param config
+   */
+  public async export(config: ExportConfig): Promise<AnyJson> {
+    this.config = this.validate(config);
+
+    const { outputDir, plan, query } = this.config;
+
+    if (outputDir) {
+      this.setupOutputDirectory(outputDir);
+    }
+
+    let queryResults: any;
```

Comment:
how did you get this `any` here? I think there's a `QueryResults` type you can import from jsforce or maybe from core

Reply:
Because I didn't want to spend too much time typing everything when there are stories to handle refactoring all this into `@salesforce/data`, and I think this is just internal.  Still useful though so I'll change it.